server/src/main/java/org/apache/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -158,6 +162,14 @@ public CoordinatorDynamicConfig(
     this.balancerComputeThreads = Math.max(balancerComputeThreads, 1);
     this.specificDataSourcesToKillUnusedSegmentsIn
         = parseJsonStringOrArray(specificDataSourcesToKillUnusedSegmentsIn);
+
+    if (null != killTaskSlotRatio) {
+      Preconditions.checkArgument(
+          killTaskSlotRatio >= 0 && killTaskSlotRatio <= 1,
+          "killTaskSlotRatio must be >= 0 and <= 1"
+      );
+    }
```

Comment:
I suggest using the new Druid exception wrapper:
```suggestion
    if (null != killTaskSlotRatio && (killTaskSlotRatio < 0 || killTaskSlotRatio > 1)) {
      throw InvalidInput.exception("killTaskSlotRatio [%.2f] is invalid. It must be >= 0 and <= 1.", killTaskSlotRatio)
    }
```